plugins/modules/vault_pki_generate_certificate.py

```diff
@@ -0,0 +1,269 @@
+      type: str
+    uri_sans:
+      description:
+        - Specifies the requested URI Subject Alternative Names.
+      type: list
+"""
+
+EXAMPLES = """
+- name: Login and use the resulting token
+  community.hashi_vault.vault_login:
+    url: https://localhost:8200
+    auth_method: ldap
+    username: "john.doe"
+    password: "{{ user_passwd }}"
+  register: login_data
+
+- name: Generate a certificate with a existing token
+  community.hashi_vault.vault_pki_generate_certificate:
+    name: test.example.org
+    common_name: test.example.org
+    ttl: 8760h
+    alt_names:
+      - test2.example.org
+      - test3.example.org
+    url: https://vault:8201
+    auth_method: token
+    token: "{{ login_data.login.auth.client_token }}"
+  register: cert_data
+
+- name: Display generated certificate
+  debug:
+    msg: "{{ cert_data.data.data.certificate }}"
+"""
+
+RETURN = """
+data:
+  description: Information about newly generated certificate
+  returned: success
+  type: complex
+  contains:
+    lease_id:
+      description: Vault lease attached to certificate
+      returned: success
+      type: string
+      sample: pki/issue/test/7ad6cfa5-f04f-c62a-d477-f33210475d05
+    renewable:
+      description: Is certificate renewable ?
+      returned: success
+      type: boolean
+      sample: false
+    lease_duration:
+      description: Vault lease duration
+      returned: success
+      type: int
+      sample: 21600
+    data:
+      description: Payload
+      returned: success
+      type: complex
+      contains:
+        certificate:
+          description: Newly generated X509 certificate
+          returned: success
+          type: string
+          sample: "-----BEGIN CERTIFICATE-----...-----END CERTIFICATE-----"
+        issuing_ca:
+          description: Certificate Authority X509 certificate
+          returned: success
+          type: string
+          sample: "-----BEGIN CERTIFICATE-----...-----END CERTIFICATE-----"
+        ca_chain:
+          description: Linked list of Certificate Authorities X509 certificates
+          returned: success
+          type: list
+          sample: ["-----BEGIN CERTIFICATE-----...-----END CERTIFICATE-----"]
+        private_key:
+          description: Private key used to generate certificate
+          returned: success
+          type: string
+          sample: "-----BEGIN RSA PRIVATE KEY-----...-----END RSA PRIVATE KEY-----"
+        private_key_type:
+          description: Private key algorithm
+          returned: success
+          type: string
+          sample: rsa
+        serial_number:
+          description: Certificate's serial number
+          returned: success
+          type: string
+          sample: 39:dd:2e:90:b7:23:1f:8d:d3:7d:31:c5:1b:da:84:d0:5b:65:31:58
+    warning:
+      description: Warning throw during generation
+      returned: success
+      type: string
+"""
+
+import traceback
+
+from ansible.errors import AnsibleError
```

Comment:
Modules cannot import from Ansible core libraries except `module_utils`, so we can't use `AnsibleError` (this exception class is used in plugins).
```suggestion
```

Reply:
Fixed in commit d7c4dc98df2ae95e50ad4da21d3be935793f6668